### ISSUE TITLE
Improve section on CSRF protection on seaside.st

### ISSUE THREAD

**Issue by theseion** [SeasideSt/Seaside#990]
From a recent discussion on seaside-dev, here are two explanations that could be useful:

1:
> CSRF is basically being able to execute a task from a different program impersonating a user that is already logged into a different site. 
>
> The attacker knows the user is logged into some program.  They know how to get that program to do something bad.  They trick the users into executing some code that does something bad on the other program that is running.  Click here to get free $50 gift card! href=https://imloggedin.com?my really bad code goes here.  This works because my really bad code is going to be sent to the location the user is logged into which means it will have access to cookies and will think this is a valid request for a user that is logged in.  Validating the cookies is not sufficient to ensure the request came from your application.
>
> You can prevent this from happening if you have something on the current page in the browser that is unique, you then check that token to make sure the request is coming from your program and not from some other program running on your computer (my really bad code).  The server needs to validate the data is coming from the browser (Header, URL, Field) matches what was set by your application (for each page) before executing any tasks.  The attacker doesn't have access to that token so trying to execute some task should fail.

2:
> CSRF usually requires a URL that can trigger an action. In Seaside, if you use continuations, the URL will contain a continuation key that specifies the state of the session (the session will usually be identified by a cookie). Every callback (action) has a key.
Since the continuation key is a random string bound to the session (multiple session could use the same continuation key without problems) an attacker would have to guess the continuation key in order to perform a CSRF. In addition callbacks are usually state dependent, i.e. specific to a page and the state to that page, so it's usually not possible to trigger a callback outside of this context.

**Comment by marschall** [comment]
I don't think we should try to explain CSRF attacks, there are better sources for this. I think we should only explain how Seaside protects against this.
